odcockpit/src/plugins/environmentviewer/EnvironmentViewerGLWidget.cpp

```diff
@@ -125,13 +125,55 @@ namespace cockpit {
                     m_velodyneSharedMemory(NULL),
                     m_hasAttachedToSharedImageMemory(false),
                     m_velodyneFrame(),
-                    m_CPCreceived(false),
+                    m_12_startingSensorID_32(0),//The first HDL-32E CPC starts from Layer 0
+                    m_11_startingSensorID_32(2),//The second HDL-32E CPC starts from Layer 2
+                    m_9_startingSensorID_32(5),//The third HDL-32E CPC starts from Layer 5
+                    m_12_cpcDistance_32(""),
+                    m_11_cpcDistance_32(""),
+                    m_9_cpcDistance_32(""),
+                    m_previousCPC32TimeStamp(0),
+                    m_cpcMask_32(0),
                     m_cpc(),
                     m_cpcMutex(),
                     m_SPCReceived(false),
                     m_CPCReceived(false),
-                    m_recordingYear(0) {}
-
+                    m_recordingYear(0) {
+                    
+                float sensorIDs_32[32];
+                bool use32IncrementA = true;
+                sensorIDs_32[0] = START_V_ANGLE_32;
+                //Derive the 32 vertical angles for HDL-32E based on the starting angle and the two increments
+                for (uint8_t counter = 1; counter < 31; counter++) {
+                    if (use32IncrementA) {
```

Comment:
Please replace this 5-line if-then-else-statement with a one-liner using the ternary operator ? :